### PR TITLE
bean: Add subcriptions with payment methods

### PR DESCRIPTION
**Pull request by nizarmah**
Joins subscriptions and payment methods

This allows us to display the following
image

Also adds horizontal lines around each row
This creates a better visual separation

Also moves the default subscription to the bottom
If you have plenty of subs, you most likely don't want to end

Testing instructions:
1. `dc down --volumes`
2. `dc up postgres -d`
3. `dc up bean`
4. `dc exec -e INTEGRATIONS=1 bean go test ./...`
5. Go to [`/subscriptions`](https://localhost:8080/subscriptions)

image

### CODE REVIEW

File: bean/internal/driver/seed/003_payment_methods.sql [hunk 1]
```diff
@@ -10,7 +10,8 @@ VALUES
 -- filler data
 ('10000000-0000-0000-0000-000000000001', '10000000-0000-0000-0000-000000000001', 'Personal virtual card', '4242', 'visa', 04, 2034),
 ('10000000-0000-0000-0000-000000000002', '10000000-0000-0000-0000-000000000001', 'Business virtual card', '5100', 'mastercard', 08, 2034),
-('10000000-0000-0000-0000-000000000003', '10000000-0000-0000-0000-000000000001', 'Personal debit card', '5556', 'visa', 12, 2034)
+('10000000-0000-0000-0000-000000000003', '10000000-0000-0000-0000-000000000001', 'Personal debit card', '5556', 'visa', 12, 2034),
+('10000000-0000-0000-0000-000000000004', '10000000-0000-0000-0000-000000000001', '', '8431', 'amex', 04, 2035)
```
Comment (nizarmah) [inline bean/internal/driver/seed/003_payment_methods.sql:14]: This is on purpose. 

We need to test PMs with empty labels.

File: bean/internal/driver/seed/004_subscriptions.sql [hunk 2]
```diff
@@ -11,12 +11,9 @@ VALUES
...
-('10000000-0000-0000-0000-000000000009', '10000000-0000-0000-0000-000000000001', '10000000-0000-0000-0000-000000000002', 'Domain - whatisbean.com', 'https://porkbun.com', 1037, 1, 'year'),
-('10000000-0000-0000-0000-000000000010', '10000000-0000-0000-0000-000000000001', '10000000-0000-0000-0000-000000000002', 'Domain - whatispear.com', 'https://porkbun.com', 1037, 1, 'year'),
-('10000000-0000-0000-0000-000000000011', '10000000-0000-0000-0000-000000000001', '10000000-0000-0000-0000-000000000002', 'HEY', 'https://hey.com', 34900, 1, 'year')
+('10000000-0000-0000-0000-000000000006', '10000000-0000-0000-0000-000000000001', '10000000-0000-0000-0000-000000000002', 'Domain - whatisbean.com', 'https://porkbun.com', 1037, 1, 'year'),
+('10000000-0000-0000-0000-000000000007', '10000000-0000-0000-0000-000000000001', '10000000-0000-0000-0000-000000000004', 'Insurance', 'https://safetywing.com', 5508, 1, 'month'),
+('10000000-0000-0000-0000-000000000008', '10000000-0000-0000-0000-000000000001', '10000000-0000-0000-0000-000000000004', 'Magazine', '', 099, 6, 'month')
```
Comment (nizarmah) [inline bean/internal/driver/seed/004_subscriptions.sql:16]: This is on purpose. 

We need to tests Subs with no providers.

Also with X months, where X > 1

File: bean/internal/adapter/handler/subscriptions/subscriptions.go [hunk 3]
```diff
@@ -63,6 +65,12 @@ func makeSubViewData(subscription *entity.Subscription) entity.SubscriptionViewD
 		Provider:  subscription.Provider,
 		Amount:    fmt.Sprintf("$%d.%02d", dollars, cents),
 		Frequency: frequency,
```
Comment (nizarmah) [inline bean/internal/adapter/handler/subscriptions/subscriptions.go:67]: The frequency is showing "Every 6 month". We need to fix that.

**Review by nizarmah — COMMENTED** [review]
Looks good

File: bean/internal/driver/template/subscriptions.html [hunk 4]
```diff
@@ -30,9 +24,20 @@ <h5>
...
+        {{ else }}
+        {{ .PaymentMethodLast4 }} &nbsp;·&nbsp; {{ .PaymentMethodExpMonth }} / {{ .PaymentMethodExpYear }}
+        {{ end }}
+      </a>
+    </p>
     <p class="actionbar">
       <label for="delete-sub-{{ .ID }}">Delete</label>
     </p>
+    <br /><br />
+    <hr />
```
Comment (nizarmah) [inline bean/internal/driver/template/subscriptions.html:40]: Switch this to be above instead of below

We will need to use the similar logic for Payment Methods
There it has to be above so we don't check if the item is the last one or not

**Review by nizarmah — COMMENTED** [review]
Actually, major change

File: bean/internal/driver/template/subscriptions.html (hunk 4 above)
Comment (nizarmah) [reply bean/internal/driver/template/subscriptions.html]: 375c98097d75f9358bf04d684a9f226bef729e98

File: bean/internal/adapter/handler/subscriptions/subscriptions.go (hunk 3 above)
Comment (nizarmah) [reply bean/internal/adapter/handler/subscriptions/subscriptions.go:67]: 375c98097d75f9358bf04d684a9f226bef729e98

**Review by nizarmah — COMMENTED** [review]
Nice!
